common/component/postgresql/awsIAM/aws.go

```diff
@@ -0,0 +1,103 @@
+package awsiam
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	aws_config "github.com/aws/aws-sdk-go-v2/config"
+	aws_credentials "github.com/aws/aws-sdk-go-v2/credentials"
+	aws_auth "github.com/aws/aws-sdk-go-v2/feature/rds/auth"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	"github.com/dapr/components-contrib/common/component/postgresql/databases"
+)
+
+type AWSIAM struct {
+	// Ignored by metadata parser because included in built-in authentication profile
+	// access key to use for accessing postgresql.
+	AWSAccessKey string `json:"awsAccessKey" mapstructure:"awsAccessKey"`
+	// secret key to use for accessing postgresql.
+	AWSSecretKey string `json:"awsSecretKey" mapstructure:"awsSecretKey"`
+	// aws session token to use.
+	AWSSessionToken string `mapstructure:"awsSessionToken"`
+	// aws region in which postgresql should create resources.
+	AWSRegion string `mapstructure:"awsRegion"`
+}
+
+func GetAccessToken(ctx context.Context, pgCfg *pgx.ConnConfig, region, accessKey, secretKey string) (string, error) {
+	dbEndpoint := pgCfg.Host + ":" + strconv.Itoa(int(pgCfg.Port))
+	var authenticationToken string
+
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.Connecting.Go.html
+	// Default to load default config through aws credentials file (~/.aws/credentials)
+	awsCfg, err := aws_config.LoadDefaultConfig(ctx)
+	if err != nil {
+		// otherwise use metadata fields
+
+		// Validate if access key and secret access key are provided
+		if accessKey == "" || secretKey == "" {
+			return "", fmt.Errorf("failed to load default configuration for AWS using accessKey and secretKey: %w", err)
+		}
+
+		// Set credentials explicitly
+		awsCfg2 := aws_credentials.NewStaticCredentialsProvider(accessKey, secretKey, "")
```

Comment:
Using `2` is generally a code smell.

```suggestion
		awsCfg = aws_credentials.NewStaticCredentialsProvider(accessKey, secretKey, "")
```